src/browser/telemetry.js

```diff
@@ -186,12 +191,21 @@ Instrumenter.prototype.instrumentNetwork = function() {
 
     replace(xhrp, 'setRequestHeader', function(orig) {
       return function(header, value) {
-        if (self.autoInstrument.networkRequestHeaders && this.__rollbar_xhr &&
-          _.isType(header, 'string') && _.isType(value, 'string')) {
-          this.__rollbar_xhr.request_headers[header] = value;
+        // If xhr.open is async, __rollbar_xhr may not be initialized yet.
+        if (!this.__rollbar_xhr) {
+          this.__rollbar_xhr = {};
         }
-        if (header.toLowerCase() === 'content-type') {
-          this.__rollbar_xhr.request_content_type = value;
+        if (_.isType(header, 'string') && _.isType(value, 'string')) {
+          if (self.autoInstrument.networkRequestHeaders) {
+            if (!this.__rollbar_xhr.request_headers) {
+              this.__rollbar_xhr.request_headers = {};
+            }
+            this.__rollbar_xhr.request_headers[header] = value;
+          }
+          // We want the content type even if request header telemetry is off.
+          if (header.toLowerCase() === 'content-type') {
+            this.__rollbar_xhr.request_content_type = value;
+          }
```

Comment:
Looks like this executes only when `(_.isType(header, 'string') && _.isType(value, 'string'))` which makes sense to me, but i believe is different than the older code. Just want to make sure that this is intended?

Reply:
Yes :+1: